xds/internal/testutils/fakeserver/server.go

```diff
@@ -61,6 +69,10 @@ type Response struct {
 // Server is a fake implementation of xDS and LRS protocols. It listens on the
 // same port for both services and exposes a bunch of channels to send/receive
 // messages.
+//
+// This server is recommended only for scenarios which cannot be tested using
+// the xDS management server (which uses envoy-go-control-plane) provided by the
+// `internal/testutils/xds/e2e` package.
```

Comment:
I made a conscious choice on this one since it is an internal test-only package and there is a good chance that when we are using it, we might miss reading the comments. So, two places is better than one :)